### PR TITLE
Compute pagination samples

### PR DESCRIPTION
**Pull request by Sita04**
Fixes #issue

> Adding samples to demonstrate pagination in GCE.

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/ListImages.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `instancesClient.close()` method on the client to
+    // safely clean up any remaining background resources.
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+
+      // Listing only non-deprecated images to reduce the size of the reply.
+      ListImagesRequest imagesRequest = ListImagesRequest.newBuilder()
+          .setProject(project)
+          .setMaxResults(2)
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/ListImages.java:55]: do we need to set this? If this example is showing the default, should we be setting parameters like this?

File: compute/cloud-client/src/main/java/compute/ListImages.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          .setMaxResults(pageSize)
+          .setFilter("deprecated.state != DEPRECATED")
+          .build();
+
+      // Use the `iteratePages` attribute of returned iterable to have more granular control of
+      // iteration over paginated results from the API. Each time you want to access the
+      // next page, the library retrieves that page from the API.
+      int pageNumber = 1;
+      for (ListPage page : imagesClient.list(imagesRequest).iteratePages()) {
+        System.out.println("Page Number: " + pageNumber++);
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/ListImages.java:94]: Shouldn't we be incrementing after this?

File: compute/cloud-client/src/main/java/compute/ListAllInstances.java [hunk 3]
```diff
@@ -39,9 +40,21 @@ public static void listAllInstances(String project) throws IOException {
     // the `instancesClient.close()` method on the client to 
     // safely clean up any remaining background resources.
     try (InstancesClient instancesClient = InstancesClient.create()) {
-      // Listing all instances for the project
```
Comment (kolea2) [inline compute/cloud-client/src/main/java/compute/ListAllInstances.java:42]: to confirm, are we replacing this example? Or should there be two?

File: compute/cloud-client/src/main/java/compute/ListImages.java (hunk 2 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/ListImages.java:94]: Yes, I am incrementing the pageNumber as part of the sysout statement!

File: compute/cloud-client/src/main/java/compute/ListAllInstances.java (hunk 3 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/ListAllInstances.java:42]: No, we are not replacing it. I felt that the comment was redundant and hence removed it.

File: compute/cloud-client/src/main/java/compute/ListImages.java (hunk 1 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/ListImages.java:55]: The idea of this snippet is to demo the abstraction involved in fetching images even though the maxResults is set. Even though, the maxResult here is set to 2 and 2 images are fetched per call, this is abstracted when the images are listed.

Comment (kolea2) [reply compute/cloud-client/src/main/java/compute/ListImages.java:55]: How many images are fetched per call without setting this? Would it be valuable to show both? I am just a little confused having one example explicitly setting `maxResults`, and one arbitrarily picking 2.

Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/ListImages.java:55]: If the method is not set, there would be a single abstracted call, which will fetch the images.
By setting setMaxResults, we are explicitly mentioning the number of images to be fetched for each page and inorder to get additional images, we can use the next page token.

But here, that complexity is abstracted in the code sample, even though the underlying mechanism will involve calling multiple next page tokens. 

The first method abstracts in entirety the pagination concept, whereas the second method shows how to iterate with the returned pages.

**Review by kolea2 — APPROVED** [review]

**Pull request merged**